plugin/pkg/scheduler/schedulercache/node_info.go

```diff
@@ -403,11 +403,26 @@ func (n *NodeInfo) updateUsedPorts(pod *v1.Pod, used bool) {
 			// "0" is explicitly ignored in PodFitsHostPorts,
 			// which is the only function that uses this value.
 			if podPort.HostPort != 0 {
+				// user does not explicitly set protocol, default is tcp
+				portProtocol := podPort.Protocol
+				if podPort.Protocol == "" {
+					portProtocol = v1.ProtocolTCP
+				}
+
+				// user does not explicitly set hostIP, default is 0.0.0.0
+				portHostIP := podPort.HostIP
+				if podPort.HostIP == "" {
+					portHostIP = util.DefaultBindAllHostIP
+				}
+
+				str := fmt.Sprintf("%s/%s/%d", portProtocol, portHostIP, podPort.HostPort)
```

Comment:
better to put this logic together with `decode`. Otherwise it looks like a magic.

Reply:
You have the same concern as @k82cn. As I explained before, I don't want to wrap this `Sprintf` logic here in another function here (need to construct a new hostPortObject and then do Sprintf, which will have less performance). Besides the api validation logic is just use `fmt.Sprintf` which you can check

Reply:
If we want to change the encode/decode method in future, it'll be really helpful to have it together. otherwise, it will be really hard. Even if you find it here, you are still not confident if there is other place that needs change.

Reply:
@anfernee I knew your concern.It's all trade-off here. I think we value performance here in scheduler.

Reply:
Besides about what you said " really hard" and "not confident", I think it is exaggerated.

Reply:
`decode` is better, but it's not a blocker comments of this PR. @anfernee , please create a new PR for it if you're interesting :).